core/server/common/src/main/java/alluxio/master/journal/AbstractCatchupThread.java

```diff
@@ -13,15 +13,20 @@
 
 import alluxio.ProcessUtils;
 
+import alluxio.thread.AutopsyThread;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
  * Abstract helper for tracking and terminating a thread for journal catch-up task.
  */
-public abstract class AbstractCatchupThread extends Thread {
+public abstract class AbstractCatchupThread extends AutopsyThread {
```

Comment:
Now both `UfsCatchupThread` and `RaftCatchupThread` will record the unhandled exception if it is killed by one.

Reply:
Note that right now there's no one checking on those kept exceptions, but we are about to add some monitors on that. The monitors are not included in this PR.